controllers/system/system_controller.go

```diff
@@ -1548,6 +1550,22 @@ func (r *SystemReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	// Cancel any existing monitors
 	r.CloudManager.CancelMonitor(instance)
 
+	// Restore the data network status
+	if r.checkRestoreInProgress(instance) {
+		r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Restoring '%s' system resource status without doing actual reconciliation", instance.Name)
+		if err := r.RestoreSystemStatus(instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		if err := r.ClearAnnotation(instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation {
```

Comment:
Need to add `instance.Status.Reconciled ` just as in platformnetwork_controller.go.

```
	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation &&
		instance.Status.Reconciled {

```

Reply:
Changes are done